faunadb-android/src/test/java/com/faunadb/client/AndroidClientSpec.java

```diff
@@ -0,0 +1,889 @@
+package com.faunadb.client;
+
+import com.faunadb.client.errors.BadRequestException;
+import com.faunadb.client.errors.NotFoundException;
+import com.faunadb.client.errors.UnauthorizedException;
+import com.faunadb.client.query.Expr;
+import com.faunadb.client.test.FaunaDBTest;
+import com.faunadb.client.types.Field;
+import com.faunadb.client.types.Value;
+import com.faunadb.client.types.Value.ObjectV;
+import com.faunadb.client.types.Value.RefV;
+import com.faunadb.client.types.Value.StringV;
+import com.faunadb.client.types.time.HighPrecisionTime;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.joda.time.LocalDate;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Arrays;
+import java.util.Random;
+
+import static com.faunadb.client.query.Language.Action.CREATE;
+import static com.faunadb.client.query.Language.Action.DELETE;
+import static com.faunadb.client.query.Language.*;
+import static com.faunadb.client.query.Language.TimeUnit.*;
+import static com.faunadb.client.types.Codec.*;
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.collection.IsArrayContainingInOrder.arrayContaining;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.collection.IsIterableContainingInOrder.contains;
+import static org.joda.time.DateTimeZone.UTC;
+import static org.junit.Assert.assertThat;
+
+public class AndroidClientSpec extends FaunaDBTest {
```

Comment:
Well, apparently this approach will work partially, because given we use `@BeforeClass` to setup the database and JUnit force this method to be static, we won't be able to call this protected method. That way we would have to duplicate at least the setup code, what do you think?